sdk/storage/azblob/zc_containerClient.go

```diff
@@ -145,17 +148,19 @@ func (c ContainerClient) GetProperties(ctx context.Context, gpo *GetPropertiesOp
 
 //// SetMetadata sets the container's metadata.
 //// For more information, see https://docs.microsoft.com/rest/api/storageservices/set-container-metadata.
-func (c ContainerClient) SetMetadata(ctx context.Context, metadata map[string]string, ac *ContainerAccessConditions) (ContainerSetMetadataResponse, error) {
-	if !ac.ModifiedAccessConditions.IfUnmodifiedSince.IsZero() || *ac.ModifiedAccessConditions.IfMatch != ETagNone || *ac.ModifiedAccessConditions.IfNoneMatch != ETagNone {
-		return ContainerSetMetadataResponse{}, errors.New("the IfUnmodifiedSince, IfMatch, and IfNoneMatch must have their default values because they are ignored by the blob service")
-	}
-	metadataOptions := ContainerSetMetadataOptions{
-		Metadata: &metadata,
-	}
-
-	mac, lac := ac.pointers()
-
-	return c.client.SetMetadata(ctx, &metadataOptions, lac, mac)
+func (c ContainerClient) SetMetadata(ctx context.Context, options *SetMetadataContainerOptions) (ContainerSetMetadataResponse, error) {
+	// TODO: Ask Ze/Adele: Why we introduced this check. We should let service do this kind of validations.
+	//if ac != nil && ac.ModifiedAccessConditions != nil {
```

Comment:
The access conditions are totally ignored, so the cx may be getting the undesired experience.